internal/controllers/alert_controller_test.go

```diff
@@ -186,8 +187,22 @@ func TestAlertReconciler_EventHandler(t *testing.T) {
 	stopCh := make(chan struct{})
 	go eventServer.ListenAndServe(stopCh, eventMdlw, store)
 
+	rcvG := g
+	var rcvMetadata map[string]string
 	rcvServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		req = r
+
+		if rcvMetadata != nil {
+			b, err := io.ReadAll(r.Body)
+			rcvG.Expect(err).ToNot(HaveOccurred())
+
+			var event eventv1.Event
+			err = json.Unmarshal(b, &event)
+			rcvG.Expect(err).ToNot(HaveOccurred())
+
+			rcvG.Expect(event.Metadata).To(BeEquivalentTo(rcvMetadata))
+		}
```

Comment:
I'm just initializing `rcvG` with the test scoped `g` because the type is pkg-private and I can't just declare a `var` with it. I'm actually updating it with the subtest scoped `g` down below

Reply:
But sure I will make it only unit then 👌

Reply:
fixed, thanks!